### PR TITLE
NEW: ICS Direct debit per each bank account

### PR DESCRIPTION
**Pull request by prietojc**
# New: ICS per each bank account. Fields moved into bank account table
Currently there is just one default ICS but if we have got several bank accounts, need store ICS (Creditor Identifier) for each account.

It needs add two new fields in bank account table: **ics** and **ics_transfer**, because some banks use different codes for direct debit and bank transfer.

Before creating a file to withdraw we need select the right bank account.

### CODE REVIEW

File: htdocs/compta/bank/class/account.class.php [hunk 1]
```diff
@@ -213,6 +214,14 @@ class Account extends CommonObject
 	 * @var int
 	 */
 	public $date_solde;
+	
+	/**
+	 * Creditor Identifier CI and Transfer ICS. Some banks use different ICS for direct debit and bank tranfer
+	 * @var string
+	 */
+	public $ics;
+	public $ics_transfer;
```
Comment (frederic34) [inline htdocs/compta/bank/class/account.class.php:223]: @prietojc please add doc on ics_transfer

Comment (prietojc) [reply htdocs/compta/bank/class/account.class.php:228]: OK

Comment (frederic34) [reply htdocs/compta/bank/class/account.class.php:228]: :+1:

File: htdocs/compta/bank/card.php [hunk 2]
```diff
@@ -94,6 +94,9 @@
 
 	$object->proprio = trim(GETPOST("proprio", 'alphanohtml'));
 	$object->owner_address   = trim(GETPOST("owner_address", 'nohtml'));
+	
+	$object->ics             = trim($_POST["ics"]);
+	$object->ics_transfer    = trim($_POST["ics_transfer"]);
```
Comment (atm-maxime) [inline htdocs/compta/bank/card.php:99]: Use the GETPOST function please

**Comment by atm-maxime** [comment]
Thanks for this contribution.
You also have to modify the SQL upgrade file to have the new fields created.

File: htdocs/compta/bank/card.php (hunk 2 above)
Comment (prietojc) [reply htdocs/compta/bank/card.php]: OK

**Comment by prietojc** [comment]
@atm-maxime Thanks for your review.

File: htdocs/compta/bank/card.php [hunk 3]
```diff
@@ -730,6 +736,14 @@
 				}
 			}
 			print '</td></tr>';
+			
+			print '<tr><td>'.$langs->trans("ICS").'</td>';
```
Comment (eldy) [inline htdocs/compta/bank/card.php:740]: If the module direct debit and credit transfer are not enabled, i think we can hide this information (most users don't know what it is related too when modules are not enabled). Donc(t you think so ?

Comment (prietojc) [reply htdocs/compta/bank/card.php:740]: Yes, I agree. I will commit.

(Happy New Year 2021!!)

File: htdocs/compta/prelevement/create.php [hunk 4]
```diff
@@ -49,6 +49,7 @@
 $action = GETPOST('action', 'aZ09');
 $mode = GETPOST('mode', 'alpha') ?GETPOST('mode', 'alpha') : 'real';
 $format = GETPOST('format', 'aZ09');
+$id_bankaccount = GETPOST ( 'id_bankaccount','int' );
```
Comment (frederic34) [inline htdocs/compta/prelevement/create.php:52]: @prietojc can you fix spaces here

**Comment by rycks** [comment]
Hello, i'm sorry but ICS seems to be unique for a company not for a Bank Account. If i have 2 or 3 bank accounts and i want to make SEPA PRE i need only one ICS. ICS is given by central bank of a country (Banque de France in France for example).

**Comment by prietojc** [comment]
In Spain ICS is different in every account.

**Pull request merged**